scripts/publish-docker-images

```diff
@@ -68,9 +69,11 @@ push_linux_amd_agent_docker_hub() {
 	# Pushing to docker hub
 	docker load < "${tarball}"
 	for image_tag in ${IMAGE_TAG_VERSION_AMD} ${IMAGE_TAG_SHA_AMD} ${IMAGE_TAG_LATEST_AMD}; do
-		tag_and_push_docker ${IMAGE_NAME} ${image_tag}
+		AMD_IMAGE_SHA=$(tag_and_push_docker ${IMAGE_NAME} ${image_tag} | tail -1 | cut -d" " -f3)
 	done
 
+	export AMD_IMAGE_SHA
```

Comment:
I'm still not a huge fan of extending this script for `source`-ing. Exporting this variable here isn't entirely necessary can be be declared in the script's top level scope - I'd prefer it be there if this going to be part of the "API".

Also, if you haven't done so already, please take a read through the script knowing that the functions and variables provided here are going to be `source`'d into the sourcing script's scope.

Things to keep in mind when sourcing:

1. Are there conflicts to behavior here that could be inherited from the caller?
 
   * Are these things protected against here, should they be? (yes, they should)

2. Are there variables that would further influence the caller? 

   * Are the callers equipped to or protected against changes here that they may or may not be aware of?

3. Are you clobbering data or functions the caller already has?

   * Default settings in this scope may be applied in such a way that overwrites what the caller has.